Branch add priority in addcommand

itsjerryho opened the pull request:
Closes #102

Commit:
Edit AddCommand to support priority

- Edit ArgumentTokenizer's findPrefixPosition method

Commit:
Edit AddCommandParserTest to support priority

Commit:
Fix checkstyle

hyngkng commented on src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:
```diff
@@ -69,7 +69,7 @@ public static ArgumentMultimap tokenize(String argsString, Prefix... prefixes) {
      * {@code fromIndex} = 0, this method returns 5.
      */
-    private static int findPrefixPosition(String argsString, String prefix, int fromIndex) {
+    protected static int findPrefixPosition(String argsString, String prefix, int fromIndex) {
```
Not sure if changing this to protected would violate anything. Is it possible to implement the priority in addCommand without changing this access modifier?

itsjerryho replied:
Implemented using regex and changed the access modifier back to private!

hyngkng reviewed (approved):
Added my comment! :)

Commit:
Edit AddCommandParser to use regex instead of ArgumentTokenizer's findPrefixPosition.

- Change findPrefix Position back to private method
- Fix Javadocs for CommandException

ChooJiaXin commented on src/main/java/seedu/address/logic/parser/AddCommandParser.java:
```diff
@@ -33,17 +36,24 @@ public AddCommand parse(String args) throws ParseException {
+        }
+
         if (isRemindPresent || isRemindTypo) {
+            // remove remind from args as ArgumentTokenizer cannot parse remind without prefix
             String argsWithoutRemind = args.replace(" remind", "");
             argMultimap = ArgumentTokenizer.tokenize(argsWithoutRemind, PREFIX_NAME, PREFIX_DEADLINE,
-                    PREFIX_MODULE_CODE);
+                    PREFIX_MODULE_CODE, PREFIX_PRIORITY);
```
What if the user inputs remind without a priority?

itsjerryho replied:
The argMultimap will just parse those that are present into the HashMap. Like it won't crash or anything

ChooJiaXin commented on src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:
```diff
@@ -68,14 +76,23 @@ public void parse_allFieldsPresent_success() {
                 + MODULE_CODE_DESC_LAB + REMIND_DESC, new AddCommand(expectedAssignmentWithRemind));
 
-        // all fields with remind
+        // TODO JERRY: Find out why an Add Command without priority passes the test case as well eg. expectedAssignment
+        // all compulsory fields with priority
+        assertParseSuccess(parser, NAME_DESC_LAB + DEADLINE_DESC_LAB + MODULE_CODE_DESC_LAB + PRIORITY_DESC,
+                new AddCommand(expectedAssignmentWithPriority));
+
+        // all compulsory fields with remind
```
Is it because Priority is an optional field? Does remind have the same problem as well or is it just Priority?

itsjerryho replied:
Yup remind has the same problem as well. I think it has something to do with how we implement .equals for the Commands in general. We kinda just check whether they're the same type I think

ChooJiaXin reviewed (commented):
Good job! Left a few questions below regarding your code, but otherwise LGTM :)

Commit:
Edit message_usage in AddCommand
